### PR TITLE
Fixed memory leak

### PR DESCRIPTION
**Pull request by Jacob-Mango**
Imagine saying you will no longer be working on DayZ mods and then making a pull request just a little over a week later on one. Couldn't be me.

### CODE REVIEW

File: JM/CF/Scripts/3_Game/CommunityFramework/Game/DayZGame.c [hunk 1]
```diff
@@ -1,5 +1,7 @@
 modded class DayZGame
 {
+	ref CF_EventRPCArgs m_CF_EventRPCArgs = new CF_EventRPCArgs();
+
```
Comment (Arkensor) [inline JM/CF/Scripts/3_Game/CommunityFramework/Game/DayZGame.c:4]: Shouln't this also be autoptr?

File: JM/CF/Scripts/1_Core/CommunityFramework/Logging/CF_Trace.c [hunk 2]
```diff
@@ -202,310 +202,372 @@ class CF_Trace : Managed
 
 static CF_Trace CF_Trace_Instance(Class instance)
 {
-	return new CF_Trace(CF_Trace.FormatInstance(instance), string.Empty, 0, true);
+	auto trace = new CF_Trace(CF_Trace.FormatInstance(instance), string.Empty, 0, true);
+	return trace;
```
Comment (Arkensor) [inline JM/CF/Scripts/1_Core/CommunityFramework/Logging/CF_Trace.c:206]: What is the idea behind this?

File: JM/CF/Scripts/3_Game/CommunityFramework/Game/DayZGame.c (hunk 1 above)
Comment (Jacob-Mango) [reply JM/CF/Scripts/3_Game/CommunityFramework/Game/DayZGame.c:4]: It works as-is with no reports of memory leaks. The problem may have been passing new instances straight into a function without the instance ever being set to a ref variable.. Never gets a chance for it's ref count to be updated when the function call ends.

File: JM/CF/Scripts/1_Core/CommunityFramework/Logging/CF_Trace.c (hunk 2 above)
Comment (Jacob-Mango) [reply JM/CF/Scripts/1_Core/CommunityFramework/Logging/CF_Trace.c:206]: Assigning to a variable so it can have it's ref count set.

**Review by Arkensor — APPROVED** [review]

**Pull request merged**